mapadroid/worker/WorkerQuests.py

```diff
@@ -841,97 +822,104 @@ def _handle_stop(self, timestamp: float):
                 self._turn_map(self._delay_add)
                 time.sleep(1)
                 self._stop_process_time = math.floor(time.time())
-                if self._open_pokestop(self._stop_process_time) is None:
+                if self._open_pokestop(self._stop_process_time) == LatestReceivedType.UNDEFINED:
                     return
                 to += 1
                 if to > 3:
                     self.logger.warning("giving up spinning after 4 tries in handle_stop loop")
 
         self.set_devicesettings_value('last_action_time', time.time())
 
-    def _wait_data_worker(self, latest, proto_to_wait_for, timestamp):
-        if latest is None:
-            self.logger.debug("Nothing received since MAD started")
-            time.sleep(0.5)
-        elif 156 in latest and latest[156].get('timestamp', 0) >= timestamp:
-            return LatestReceivedType.GYM
+    def _check_for_data_content(self, latest, proto_to_wait_for: int, timestamp: float) \
+            -> Tuple[LatestReceivedType, Optional[Union[dict, FortSearchResultTypes]]]:
+        type_of_data_found: LatestReceivedType = LatestReceivedType.UNDEFINED
+        data_found: Optional[object] = None
+        # Check if we have clicked a gym or mon...
+        if 156 in latest and latest[156].get('timestamp', 0) >= timestamp:
```

Comment:
We may want to use enums for this. I have no idea what 156 equates to (its easy to infer but would improve readability)